### PR TITLE
Rental service APP - first version

### PR DESCRIPTION
**Pull request by Gdiazdiaz**
## Tasks completed

[Backend PR](https://github.com/PrangonGhose/Earthbnb-backend/pull/18)

- Install bootstrap for desing.
- Install axios for GET, POST and DELETE requests.
- Install react-datepicker For the Reserve function.

- Create Navbar
  - The Navbar has links to allow the user to navigate to Main page, add a porpertyy, delete a property, add a reservation, all reservations, and logout.
  - The navbar contains the icons for social networks.
  - The navbar containes the license.

- Create the components for the UI.
  - Create a launch page to allow the user to create an account or login.
    - The user can Sign in or Sign up using the username.
 
  - Create Main page to display all properties available.
    - The page has left/right arrow to navigate the list of properites.
    - Each property tab works as a link to the details page of that property.
 
  - Create page to allow the user to add a new house.
    - The page allows the user to add a new property byfilling all boxes.
    - When the property gets created succesfully, a notifivation on top appears.
  
  - Create page to allow the user to delete a house from the list.
    - The page displays all the properties, each with a delete button to destroy the item.
    - The state refreshes and the page only displays the properties that have not bean deleted.
  
  - Create page to allow the user to reserve a property for determinated dates.
    - The page allows the user to reserve a property by selecting a check in date, check out date and the house to reserve.
  
  - Create page to allow the user to see all reservations made.
    - The page displays all properties reserved

- Every page is responsive.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  "private": true,
+  "version": "0.0.0",
+  "type": "module",
+  "scripts": {
+    "dev": "vite",
+    "build": "vite build",
+    "lint": "eslint src --ext js,jsx --report-unused-disable-directives --max-warnings 0",
+    "preview": "vite preview",
+    "eslint": "npx eslint \"**/*.{js,jsx}\"",
+    "stylelint": "npx stylelint \"**/*.{css,scss}\""
```
Comment (vichuge) [inline package.json]: - It looks like there isn't a way to run tests, something like `npm run test`, I kindly invite you to add the commando to run tests properly on the app. 🐱

**Review by vichuge — COMMENTED** [review]
Hi dear student, 

Good job so far!
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!

mario_ladders


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by PrangonGhose** [comment]
### Changes made after first review
- Added the functionality so that `npm run test` can run. But adding tests were not a requirement in this project and so no tests are added in this scope.

**Review by Meltrust — APPROVED** [review]
#### Hi @Gdiazdiaz,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Beautiful, gorgeous frontend✔️ 
- Nice code structure ✔️ 
- Superb mobile implementation ✔️ 
- App is working well✔️ 
- Wonderful job✔️ 



Cheers and Happy coding!👏👏👏


------

**Pull request merged**